admiral/pkg/clusters/serviceentry.go

```diff
@@ -484,29 +514,45 @@ func AddServiceEntriesWithDr(ctx context.Context, rr *RemoteRegistry, sourceClus
 
 				//clean service entry in case no endpoints are configured or if all the endpoints are invalid
 				if (len(seDr.ServiceEntry.Endpoints) == 0) || deleteOldServiceEntry {
-					deleteServiceEntry(ctx, oldServiceEntry, syncNamespace, rc)
-					cache.SeClusterCache.Delete(seDr.ServiceEntry.Hosts[0])
-					// after deleting the service entry, destination rule also need to be deleted if the service entry host no longer exists
-					deleteDestinationRule(ctx, oldDestinationRule, syncNamespace, rc)
+					if !skipSEUpdate {
+						deleteServiceEntry(ctx, oldServiceEntry, syncNamespace, rc)
+						cache.SeClusterCache.Delete(seDr.ServiceEntry.Hosts[0])
+					}
+					if !skipDRUpdate {
+						// after deleting the service entry, destination rule also need to be deleted if the service entry host no longer exists
+						deleteDestinationRule(ctx, oldDestinationRule, syncNamespace, rc)
+					}
 				} else {
-					//nolint
-					newServiceEntry := createServiceEntrySkeletion(*seDr.ServiceEntry, seDr.SeName, syncNamespace)
-					if newServiceEntry != nil {
-						newServiceEntry.Labels = map[string]string{common.GetWorkloadIdentifier(): fmt.Sprintf("%v", identityId)}
-						addUpdateServiceEntry(ctx, newServiceEntry, oldServiceEntry, syncNamespace, rc)
-						cache.SeClusterCache.Put(newServiceEntry.Spec.Hosts[0], rc.ClusterID, rc.ClusterID)
+					if !skipSEUpdate {
+						//nolint
+						newServiceEntry := createServiceEntrySkeletion(*seDr.ServiceEntry, seDr.SeName, syncNamespace)
+						if newServiceEntry != nil {
+							newServiceEntry.Labels = map[string]string{common.GetWorkloadIdentifier(): fmt.Sprintf("%v", identityId)}
+							addUpdateServiceEntry(ctx, newServiceEntry, oldServiceEntry, syncNamespace, rc)
```

Comment:
Is it possible to move the logic to skip updating into `addUpdateServiceEntry` and `addUpdateDestinationRule`? That might make it cleaner.

Reply:
hey @aattuluri , I agree it will make it cleaner but I would need to do the same for the `deleteSE` and `deleteDR` funcs. Plus, in order to avoid adding it to the SEClusterCache, I added the logic here.

Reply:
Got it, I am good then.